Data-converge-revised

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# vaspup2.0 - Seán Kavanagh (sean.kavanagh.19@ucl.ac.uk), 2023
+# revised by Xiangchao Meng (xiangchao.meng.22@ucl.ac.uk), 08/10/2023
+rm -f data # delete data file if exists
+
+n=0
+shopt -s extglob
+custom_tab='  '
+
+# The bc command in line 43 is not compatible with the out put of checkforce command in sci-form, i.e., bc can't recognise output as 1e-05. This sci2dec() function was added to convert sci-form to numerical form.
+sci2dec() {
+    printf "%.10f" "$1"
+}
+
+if ! [ -x "$(command -v checkforce)" ]
+then 
+    echo -e "checkforce (from https://github.com/bjmorgan/VASP-Utilities) is not on PATH, will not check force convergence."
+fi
+
+for i in {!(*_*),*_*}/CONTCAR; do
+    if ! [[ $i == "*_*/CONTCAR" ]]
+    then
+    if ! [[ $i == "!(*_*)/CONTCAR" ]]
+    then
+    if [[ $i == *k_* ]]; then
+        spacing="\t"
+    else
+        spacing="\t\t"
+    fi
+
+    folder=${i/CONTCAR/}
+    name=${folder//[\/]/} # folder name without slashes
+
+    n_atoms=$(sed '7q;d' $i | awk '{sum=0; for (i=1; i<=NF; i++) { sum+= $i } print sum}') # sum over row 7 of the CONTCAR
+    energy_total=$(grep TOTEN $folder/OUTCAR | awk '{print $5}' | tail -1 | tr -d '\n') # tr is to remove the trailing newline
+    energy_per_atom=$(echo "scale = 7; $energy_total / $n_atoms" | bc)
+    nelm=$(grep NELM $folder/OUTCAR | awk '{print $3}' | head -1 | tr -d ";" )
+    n_iter=$(grep Iteration $folder/OUTCAR | awk '{print $4}' | tail -1 | tr -d ")" )
+    
+    if [ -x "$(command -v checkforce)" ] && [ "$n" -ne "0" ]
+    then 
+        residual_force=$(sci2dec $("checkforce" -c 0 $folder/OUTCAR | awk '/remainder/ {print $NF}'))
+        force_diff=$(echo "scale = 7; ($force_prev - $residual_force) * 1000" | bc)  # meV/Angstrom averaged
```

Comment:
Here, the `bc` command is the problem. With the help of `sci2dec()`, there will be no syntax error.